applications/vanilla/modules/class.usercommentsmodule.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * User comments module.
+ *
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
+ * @package Vanilla
+ * @since 2.3
+ */
+
+/**
+ * Renders recently posted comments from a specific user
+ */
+class UserCommentsModule extends Gdn_Module {
+
+    /** @var int Display limit. */
+    public $limit = 10;
+
+    /** @var  int The ID of the user for whom you are showing the comments. */
+    public $userID = null;
+
+    /**
+     *
+     *
+     * @throws Exception
+     */
+    public function __construct($sender) {
+        parent::__construct();
+        $this->_ApplicationFolder = 'vanilla';
+        $this->fireEvent('Init');
+        //If you are being executed from the profile controller, get the UserID.
+        if (strtolower(val('ControllerName', $sender)) === 'profilecontroller') {
+            $this->userID = $sender->User->UserID;
```

Comment:
Instead of assuming this property exists, try using the `valr` function. Example:

```php
$this->userID = valr('User.UserID', $sender);
```